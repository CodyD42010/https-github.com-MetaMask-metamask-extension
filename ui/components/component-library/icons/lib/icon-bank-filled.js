import * as React from 'react';
import PropTypes from 'prop-types';
import { SIZES } from '../../../../helpers/constants/design-system';
import Box from '../../../ui/box/box';
import { BaseIcon } from '../../base-icon';
export const IconBankFilled = ({ size, color, className, ...props }) => {
  return React.createElement(
    BaseIcon,
    {
      size,
      color,
      className,
      ...props,
    },
    <path d="M456 399v62H47v-62c0-11 9-20 20-20h369c11 0 20 9 20 20zM149 236h-41v143h41zm82 0h-41v143h41zm82 0h-41v143h41zm82 0h-41v143h41zm82 240H26c-8 0-15-7-15-15 0-9 7-16 15-16h451c8 0 15 7 15 16 0 8-7 15-15 15zm-34-348L259 54c-4-1-11-1-15 0L60 128c-8 3-13 11-13 19v68c0 11 9 21 20 21h369c11 0 20-10 20-21v-68c0-8-5-16-13-19zm-191 56c-17 0-31-13-31-30s14-31 31-31 30 14 30 31-13 30-30 30z" />,
  );
};
IconBankFilled.propTypes = {
  /**
   * The size of the BaseIcon.
   * Possible values could be 'xxs', 'xs', 'sm', 'md', 'lg', 'xl',
   */
  size: PropTypes.oneOf(Object.values(SIZES)),

  /**
   * The color of the icon. Defaults to 'inherit'.
   */
  color: PropTypes.string,

  /**
   * An additional class name to apply to the icon.
   */
  className: PropTypes.string,

  /**
   * BaseIcon accepts all the props from Box
   */
  ...Box.propTypes,
};
