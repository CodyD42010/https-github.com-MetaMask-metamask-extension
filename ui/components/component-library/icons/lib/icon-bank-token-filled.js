import * as React from 'react';
import PropTypes from 'prop-types';
import { SIZES } from '../../../../helpers/constants/design-system';
import Box from '../../../ui/box/box';
import { BaseIcon } from '../../base-icon';
export const IconBankTokenFilled = ({ size, color, className, ...props }) => {
  return React.createElement(
    BaseIcon,
    {
      size,
      color,
      className,
      ...props,
    },
    <path d="M22 15c0 4-3 7-7 7l1-2M2 9c0-4 3-7 7-7L8 4m223 451H26v-31c0-5 5-10 10-10h185c5 0 10 5 10 10zm-6-166-93-37c-2-1-5-1-7 0l-92 37c-4 1-7 5-7 9v34c0 6 5 11 10 11h185c5 0 10-5 10-11v-34c0-4-3-8-6-9zm-76 8c0 11-9 20-20 20-12 0-21-9-21-20s9-21 21-21c11 0 20 10 20 21zm82 158v8c4 0 8-4 8-8zm-205 0h-7c0 4 3 8 7 8zm13-41c0 4 3 8 8 8 4 0 7-4 7-8zm15-71c0-5-3-8-7-8-5 0-8 3-8 8zm26 71c0 4 3 8 8 8 4 0 7-4 7-8zm15-71c0-5-3-8-7-8-5 0-8 3-8 8zm26 71c0 4 3 8 8 8 4 0 7-4 7-8zm15-71c0-5-3-8-7-8-5 0-8 3-8 8zm26 71c0 4 3 8 8 8 4 0 7-4 7-8zm15-71c0-5-3-8-7-8-5 0-8 3-8 8zm26 71c0 4 3 8 8 8 4 0 7-4 7-8zm15-71c0-5-3-8-7-8-5 0-8 3-8 8zM16 447c-4 0-8 4-8 8s4 8 8 8zm225 16c4 0 8-4 8-8s-4-8-8-8zM132 252l-2 7zm93 37 2-7zm-100-37 3 7zm-92 37-3-7zm198 158H26v16h205zm-197 8v-31H19v31zm0-31c0-1 1-2 2-2v-15c-9 0-17 8-17 17zm2-2h185v-15H36zm185 0c1 0 2 1 2 2h16c0-9-8-17-18-17zm2 2v31h16v-31zM54 414v-71H39v71zm41 0v-71H80v71zm41 0v-71h-15v71zm41 0v-71h-15v71zm41 0v-71h-15v71zM16 463h225v-16H16zm114-204 92 37 5-14-92-37zm-2 0h2l5-14c-2-1-4-1-6-1s-5 0-7 1zm-92 37 92-37-6-14-92 37zm-2 2s0-1 1-1v-1h1l-6-14c-4 1-6 4-8 6-2 3-3 7-3 10zm0 34v-34H19v34zm2 3c-1 0-2-1-2-3H19c0 10 8 18 17 18zm185 0H36v15h185zm2-3c0 2-1 3-2 3v15c10 0 18-8 18-18zm0-34v34h16v-34zm-1-2 1 1v1h16c0-3-2-7-3-10-2-2-5-5-9-6zm-93 29c15 0 28-13 28-28h-16c0 7-5 13-12 13zm-29-28c0 15 13 28 29 28v-15c-7 0-13-6-13-13zm29-28c-16 0-29 12-29 28h16c0-7 6-13 13-13zm28 28c0-16-13-28-28-28v15c7 0 12 6 12 13zm107-163 65-28c3-2 6-2 10 0l65 28c6 3 11-4 7-9l-68-83c-4-6-12-6-16 0l-69 83c-5 5 0 12 6 9zm0 80 65 28c3 2 6 2 10 0l65-28c6-3 11 4 7 9l-68 84c-4 5-12 5-16 0l-69-84c-5-5 0-12 6-9zm69-75-71 35 71 36 71-36z" />,
  );
};
IconBankTokenFilled.propTypes = {
  /**
   * The size of the BaseIcon.
   * Possible values could be 'xxs', 'xs', 'sm', 'md', 'lg', 'xl',
   */
  size: PropTypes.oneOf(Object.values(SIZES)),

  /**
   * The color of the icon. Defaults to 'inherit'.
   */
  color: PropTypes.string,

  /**
   * An additional class name to apply to the icon.
   */
  className: PropTypes.string,

  /**
   * BaseIcon accepts all the props from Box
   */
  ...Box.propTypes,
};
