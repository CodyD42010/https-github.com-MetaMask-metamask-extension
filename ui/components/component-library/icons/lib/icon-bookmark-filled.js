import * as React from 'react';
import PropTypes from 'prop-types';
import { SIZES } from '../../../../helpers/constants/design-system';
import Box from '../../../ui/box/box';
import { BaseIcon } from '../../base-icon';
export const IconBookmarkFilled = ({ size, color, className, ...props }) => {
  return React.createElement(
    BaseIcon,
    {
      size,
      color,
      className,
      ...props,
    },
    <path d="M456 170v172c0 66-53 119-119 119H166c-66 0-119-53-119-119V170c0-51 21-88 58-106 13-6 29 4 29 19v182c0 24 9 43 26 53s39 8 61-5l27-16c1-1 6-1 7-1l27 17c15 8 27 11 38 11 10 0 18-3 23-6 17-10 26-29 26-53V83c0-15 16-25 29-19 37 18 58 55 58 106zM318 51c11 0 21 9 21 21v193c0 13-4 23-11 26-7 5-18 2-30-5l-27-16c-11-6-28-6-39 0l-27 16c-12 7-23 9-30 5-7-3-11-13-11-26V72c0-12 10-21 21-21z" />,
  );
};
IconBookmarkFilled.propTypes = {
  /**
   * The size of the BaseIcon.
   * Possible values could be 'xxs', 'xs', 'sm', 'md', 'lg', 'xl',
   */
  size: PropTypes.oneOf(Object.values(SIZES)),

  /**
   * The color of the icon. Defaults to 'inherit'.
   */
  color: PropTypes.string,

  /**
   * An additional class name to apply to the icon.
   */
  className: PropTypes.string,

  /**
   * BaseIcon accepts all the props from Box
   */
  ...Box.propTypes,
};
