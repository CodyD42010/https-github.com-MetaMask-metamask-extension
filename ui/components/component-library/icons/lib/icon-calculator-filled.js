import * as React from 'react';
import PropTypes from 'prop-types';
import { SIZES } from '../../../../helpers/constants/design-system';
import Box from '../../../ui/box/box';
import { BaseIcon } from '../../base-icon';
export const IconCalculatorFilled = ({ size, color, className, ...props }) => {
  return React.createElement(
    BaseIcon,
    {
      size,
      color,
      className,
      ...props,
    },
    <path d="M333 51H170C113 51 67 97 67 154v204c0 57 46 103 103 103h163c57 0 103-46 103-103V154c0-57-46-103-103-103zM187 392c-4 4-9 6-14 6-6 0-11-2-15-6s-6-9-6-15c0-5 2-10 6-14 2-2 4-4 7-5 5-2 11-2 16 0 1 1 2 1 3 2l3 3c4 4 6 9 6 14 0 6-2 11-6 15zm-35-97c0-2 1-5 2-7 1-3 2-5 4-7 5-5 12-7 19-6 1 0 2 1 4 1 1 1 2 2 3 2l3 3c2 2 4 4 5 7 1 2 1 5 1 7 0 6-2 11-6 15s-9 6-14 6c-3 0-6-1-8-2-3-1-5-2-7-4-4-4-6-9-6-15zm117 97c-2 2-4 3-6 4-3 1-6 2-8 2-6 0-11-2-15-6s-6-9-6-15c0-1 0-2 1-4 0-1 0-2 1-4 0-1 1-2 2-3 0-1 1-2 2-3 2-2 4-4 7-5 7-3 16-1 22 5 4 4 6 9 6 14 0 6-2 11-6 15zm0-82c-4 4-9 6-14 6-6 0-11-2-15-6s-6-9-6-15c0-5 2-10 6-14 8-8 22-8 29 0 2 2 4 4 5 7 1 2 1 5 1 7 0 6-2 11-6 15zm-79-86c-21 0-38-17-38-38v-21c0-21 17-38 38-38h123c21 0 38 17 38 38v21c0 21-17 38-38 38zm161 168c-4 4-9 6-14 6-3 0-6-1-8-2-3-1-5-2-7-4-4-4-6-9-6-15 0-5 2-10 6-14 6-6 15-8 23-5 2 1 4 3 6 5 4 4 6 9 6 14 0 6-2 11-6 15zm5-89c-1 3-3 5-5 7-4 4-9 6-14 6-6 0-11-2-15-6s-6-9-6-15c0-5 2-10 6-14 8-8 21-8 29 0 4 4 6 9 6 14 0 3 0 6-1 8z" />,
  );
};
IconCalculatorFilled.propTypes = {
  /**
   * The size of the BaseIcon.
   * Possible values could be 'xxs', 'xs', 'sm', 'md', 'lg', 'xl',
   */
  size: PropTypes.oneOf(Object.values(SIZES)),

  /**
   * The color of the icon. Defaults to 'inherit'.
   */
  color: PropTypes.string,

  /**
   * An additional class name to apply to the icon.
   */
  className: PropTypes.string,

  /**
   * BaseIcon accepts all the props from Box
   */
  ...Box.propTypes,
};
