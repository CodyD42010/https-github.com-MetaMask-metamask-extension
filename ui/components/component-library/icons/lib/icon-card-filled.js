import * as React from 'react';
import PropTypes from 'prop-types';
import { SIZES } from '../../../../helpers/constants/design-system';
import Box from '../../../ui/box/box';
import { BaseIcon } from '../../base-icon';
export const IconCardFilled = ({ size, color, className, ...props }) => {
  return React.createElement(
    BaseIcon,
    {
      size,
      color,
      className,
      ...props,
    },
    <path d="M456 165c0 13-11 24-24 24H71c-13 0-24-11-24-24 0-47 38-85 84-85h240c47 0 85 38 85 85zM47 245v102c0 47 38 85 84 85h240c47 0 85-38 85-85V245c0-14-11-25-24-25H71c-13 0-24 11-24 25zm123 119h-41c-9 0-16-7-16-16 0-8 7-15 16-15h41c8 0 15 7 15 15 0 9-7 16-15 16zm133 0h-82c-9 0-16-7-16-16 0-8 7-15 16-15h82c8 0 15 7 15 15 0 9-7 16-15 16z" />,
  );
};
IconCardFilled.propTypes = {
  /**
   * The size of the BaseIcon.
   * Possible values could be 'xxs', 'xs', 'sm', 'md', 'lg', 'xl',
   */
  size: PropTypes.oneOf(Object.values(SIZES)),

  /**
   * The color of the icon. Defaults to 'inherit'.
   */
  color: PropTypes.string,

  /**
   * An additional class name to apply to the icon.
   */
  className: PropTypes.string,

  /**
   * BaseIcon accepts all the props from Box
   */
  ...Box.propTypes,
};
