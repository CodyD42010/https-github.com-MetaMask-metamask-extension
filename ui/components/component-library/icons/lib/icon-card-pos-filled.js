import * as React from 'react';
import PropTypes from 'prop-types';
import { SIZES } from '../../../../helpers/constants/design-system';
import Box from '../../../ui/box/box';
import { BaseIcon } from '../../base-icon';
export const IconCardPosFilled = ({ size, color, className, ...props }) => {
  return React.createElement(
    BaseIcon,
    {
      size,
      color,
      className,
      ...props,
    },
    <path d="M456 445c0 9-7 16-15 16H62c-8 0-15-7-15-16 0-8 7-15 15-15h379c8 0 15 7 15 15zM321 103 101 323c-8 8-22 8-30 0-29-29-29-75 0-104L217 73c29-29 75-29 104 0 8 8 8 22 0 30zm111 81-62-62c-9-9-22-9-30 0L120 342c-9 8-9 21 0 30l62 62c29 29 75 29 104 0l146-146c29-29 29-75 0-104zM267 369l-25 25c-5 5-13 5-19 0-5-5-5-13 0-19l25-25c5-5 14-5 19 0s5 14 0 19zm81-81-50 50c-5 5-13 5-18 0-6-5-6-14 0-19l50-50c5-5 13-5 18 0 6 5 6 14 0 19z" />,
  );
};
IconCardPosFilled.propTypes = {
  /**
   * The size of the BaseIcon.
   * Possible values could be 'xxs', 'xs', 'sm', 'md', 'lg', 'xl',
   */
  size: PropTypes.oneOf(Object.values(SIZES)),

  /**
   * The color of the icon. Defaults to 'inherit'.
   */
  color: PropTypes.string,

  /**
   * An additional class name to apply to the icon.
   */
  className: PropTypes.string,

  /**
   * BaseIcon accepts all the props from Box
   */
  ...Box.propTypes,
};
