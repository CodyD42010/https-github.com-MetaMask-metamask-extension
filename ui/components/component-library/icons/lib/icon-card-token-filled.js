import * as React from 'react';
import PropTypes from 'prop-types';
import { SIZES } from '../../../../helpers/constants/design-system';
import Box from '../../../ui/box/box';
import { BaseIcon } from '../../base-icon';
export const IconCardTokenFilled = ({ size, color, className, ...props }) => {
  return React.createElement(
    BaseIcon,
    {
      size,
      color,
      className,
      ...props,
    },
    <path d="M252 332v2c0 6-5 10-11 10H57c-6 0-10-4-10-10v-2c0-36 9-45 45-45h114c37 0 46 9 46 45zM57 365c-6 0-10 4-10 10v41c0 36 9 45 45 45h114c37 0 46-9 46-45v-41c0-6-5-10-11-10zM22 15c0 4-3 7-7 7l1-2M2 9c0-4 3-7 7-7L8 4m256 130 65-28c3-2 6-2 9 0l65 28c6 3 11-4 7-9l-68-83c-5-6-13-6-17 0l-68 83c-4 5 1 12 7 9zm0 80 65 29c3 1 6 1 9 0l65-29c6-3 11 4 7 9l-68 83c-5 6-12 6-17 0l-68-83c-4-5 1-12 7-9zm66-74-56 28c-5 2-5 10 0 12l56 28c2 1 5 1 7 0l56-28c5-2 5-10 0-12l-56-28c-2-1-5-1-7 0z" />,
  );
};
IconCardTokenFilled.propTypes = {
  /**
   * The size of the BaseIcon.
   * Possible values could be 'xxs', 'xs', 'sm', 'md', 'lg', 'xl',
   */
  size: PropTypes.oneOf(Object.values(SIZES)),

  /**
   * The color of the icon. Defaults to 'inherit'.
   */
  color: PropTypes.string,

  /**
   * An additional class name to apply to the icon.
   */
  className: PropTypes.string,

  /**
   * BaseIcon accepts all the props from Box
   */
  ...Box.propTypes,
};
