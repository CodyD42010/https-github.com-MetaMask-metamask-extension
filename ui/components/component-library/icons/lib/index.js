export { IconAddSquareFilled } from './icon-add-square-filled';
export { IconBankFilled } from './icon-bank-filled';
export { IconBankTokenFilled } from './icon-bank-token-filled';
export { IconBookmarkFilled } from './icon-bookmark-filled';
export { IconCalculatorFilled } from './icon-calculator-filled';
export { IconCardFilled } from './icon-card-filled';
export { IconCardPosFilled } from './icon-card-pos-filled';
export { IconCardTokenFilled } from './icon-card-token-filled';
export { IconCategoryFilled } from './icon-category-filled';
export { IconChartFilled1 } from './icon-chart-filled-1';
export { IconChartFilled } from './icon-chart-filled';
export { IconCheckBoxOffOutline } from './icon-check-box-off-outline';
export { IconCheckBoxOnFilled } from './icon-check-box-on-filled';
export { IconCheckCircleOnFilled } from './icon-check-circle-on-filled';
export { IconClockFilled } from './icon-clock-filled';
export { IconCloseOutline } from './icon-close-outline';
export { IconCodeCircleFilled } from './icon-code-circle-filled';
export { IconCoinFilled } from './icon-coin-filled';
export { IconConnectFilled } from './icon-connect-filled';
export { IconCopyFilled } from './icon-copy-filled';
export { IconCopySuccessFilled } from './icon-copy-success-filled';
export { IconDangerFilled } from './icon-danger-filled';
export { IconDarkFilled } from './icon-dark-filled';
export { IconDataFilled } from './icon-data-filled';
export { IconDocumentCodeFilled } from './icon-document-code-filled';
export { IconEditFilled } from './icon-edit-filled';
export { IconEraserFilled } from './icon-eraser-filled';
export { IconExploreFilled } from './icon-explore-filled';
export { IconEyeFilled } from './icon-eye-filled';
export { IconEyeSlashFilled } from './icon-eye-slash-filled';
export { IconFilterOutline } from './icon-filter-outline';
export { IconFlagFilled } from './icon-flag-filled';
export { IconFlashFilled } from './icon-flash-filled';
export { IconFlashSlashFilled } from './icon-flash-slash-filled';
export { IconFullCircleFilled } from './icon-full-circle-filled';
export { IconGasFilled } from './icon-gas-filled';
export { IconGlobalFilled } from './icon-global-filled';
export { IconGlobalSearchFilled } from './icon-global-search-filled';
export { IconGraphFilled } from './icon-graph-filled';
export { IconHeartFilled } from './icon-heart-filled';
export { IconHierarchyFilled } from './icon-hierarchy-filled';
export { IconHomeFilled } from './icon-home-filled';
export { IconInfoFilled } from './icon-info-filled';
export { IconKeyFilled } from './icon-key-filled';
export { IconLightFilled } from './icon-light-filled';
export { IconLink2Outline } from './icon-link-2-outline';
export { IconLoadingFilled } from './icon-loading-filled';
export { IconLockCircleFilled } from './icon-lock-circle-filled';
export { IconLockFilled } from './icon-lock-filled';
export { IconLockSlashFilled } from './icon-lock-slash-filled';
export { IconMenuOutline } from './icon-menu-outline';
export { IconMessageQuestionFilled } from './icon-message-question-filled';
export { IconMessagesFilled } from './icon-messages-filled';
export { IconMinusSquareFilled } from './icon-minus-square-filled';
export { IconMobileFilled } from './icon-mobile-filled';
export { IconMoneyFilled } from './icon-money-filled';
export { IconMonitorFilled } from './icon-monitor-filled';
export { IconMoreHorizontalOutline } from './icon-more-horizontal-outline';
export { IconMoreVerticalOutline } from './icon-more-vertical-outline';
export { IconNotificationCircleFilled } from './icon-notification-circle-filled';
export { IconNotificationFilled } from './icon-notification-filled';
export { IconPasswordCheckFilled } from './icon-password-check-filled';
export { IconPeopleFilled } from './icon-people-filled';
export { IconProgrammingArrowsFilled } from './icon-programming-arrows-filled';
export { IconQuestionFilled } from './icon-question-filled';
export { IconReceivedOutline } from './icon-received-outline';
export { IconScanBarcodeFilled } from './icon-scan-barcode-filled';
export { IconScanFilled } from './icon-scan-filled';
export { IconScrollFilled } from './icon-scroll-filled';
export { IconSearchFilled } from './icon-search-filled';
export { IconSecurityCardFilled } from './icon-security-card-filled';
export { IconSecurityCrossFilled } from './icon-security-cross-filled';
export { IconSecurityKeyFilled } from './icon-security-key-filled';
export { IconSecuritySearchFilled } from './icon-security-search-filled';
export { IconSecuritySlashFilled } from './icon-security-slash-filled';
export { IconSecurityTickFilled } from './icon-security-tick-filled';
export { IconSecurityTimeFilled } from './icon-security-time-filled';
export { IconSecurityUserFilled } from './icon-security-user-filled';
export { IconSend1Filled } from './icon-send-1-filled';
export { IconSend2Outline } from './icon-send-2-outline';
export { IconSettingFilled } from './icon-setting-filled';
export { IconSlashFilled } from './icon-slash-filled';
export { IconSnapsFilled } from './icon-snaps-filled';
export { IconSnapsMobileFilled } from './icon-snaps-mobile-filled';
export { IconSnapsPlusFilled } from './icon-snaps-plus-filled';
export { IconSpeedometerFilled } from './icon-speedometer-filled';
export { IconStudentFilled } from './icon-student-filled';
export { IconSwapHorizontalOutline } from './icon-swap-horizontal-outline';
export { IconSwapVerticalFilled } from './icon-swap-vertical-filled';
export { IconTagFilled } from './icon-tag-filled';
export { IconTimerFilled } from './icon-timer-filled';
export { IconTrashFilled } from './icon-trash-filled';
export { IconTrendDownFilled } from './icon-trend-down-filled';
export { IconTrendUpFilled } from './icon-trend-up-filled';
export { IconUserAddFilled1 } from './icon-user-add-filled-1';
export { IconUserAddFilled } from './icon-user-add-filled';
export { IconUserCircleAddFilled } from './icon-user-circle-add-filled';
export { IconUserCircleFilled } from './icon-user-circle-filled';
export { IconUserFilled } from './icon-user-filled';
export { IconUserMinusFilled } from './icon-user-minus-filled';
export { IconUserRemoveFilled } from './icon-user-remove-filled';
export { IconUserSearchFilled } from './icon-user-search-filled';
export { IconUserTickFilled } from './icon-user-tick-filled';
export { IconWalletCardFilled } from './icon-wallet-card-filled';
export { IconWalletFilled } from './icon-wallet-filled';
export { IconWalletMoneyFilled } from './icon-wallet-money-filled';
export { IconWarningFilled } from './icon-warning-filled';
