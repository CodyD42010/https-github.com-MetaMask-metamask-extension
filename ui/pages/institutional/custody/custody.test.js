import React from 'react';
import configureMockStore from 'redux-mock-store';
import { fireEvent, waitFor, screen } from '@testing-library/react';
import thunk from 'redux-thunk';
import { renderWithProvider } from '../../../../test/lib/render-helpers';
import CustodyPage from '.';

const mockedReturnedValue = jest.fn().mockReturnValue({ type: 'TYPE' });
const mockedGetCustodianJWTList = jest.fn().mockReturnValue({ type: 'TYPE' });

const mockedGetCustodianAccounts = jest.fn().mockReturnValue(async () => null);
const mockedGetCustodianToken = jest.fn().mockReturnValue('testJWT');

const mockedGetCustodianConnectRequest = jest.fn().mockReturnValue({
  type: 'TYPE',
  custodian: 'saturn',
  token: 'token',
  apiUrl: 'url',
  custodianType: 'JSON-RPC',
  custodianName: 'Saturn',
});

jest.mock('../../../store/institutional/institution-background', () => ({
  mmiActionsFactory: () => ({
    getCustodianConnectRequest: mockedGetCustodianConnectRequest,
    getCustodianToken: mockedGetCustodianToken,
    getCustodianAccounts: mockedGetCustodianAccounts,
    getCustodianAccountsByAddress: mockedReturnedValue,
    getCustodianJWTList: mockedGetCustodianJWTList,
    connectCustodyAddresses: mockedReturnedValue,
  }),
}));

describe('CustodyPage', function () {
  const mockStore = {
    metamask: {
      providerConfig: { chainId: 0x1, type: 'test' },
      mmiConfiguration: {
        portfolio: {
          enabled: true,
          url: 'https://portfolio.io',
        },
        custodians: [
          {
            type: 'Saturn',
            name: 'saturn',
            apiUrl: 'https://saturn-custody.dev.metamask-institutional.io',
            iconUrl:
              'https://saturn-custody-ui.dev.metamask-institutional.io/saturn.svg',
            displayName: 'Saturn Custody',
            production: true,
            refreshTokenUrl: null,
            isNoteToTraderSupported: false,
            version: 1,
          },
        ],
      },
      preferences: {
        useNativeCurrencyAsPrimaryCurrency: true,
      },
      appState: {
        isLoading: false,
      },
      history: {
        mostRecentOverviewPage: '/',
      },
    },
  };

  const store = configureMockStore([thunk])(mockStore);

  it('renders CustodyPage', async () => {
    const { container } = renderWithProvider(<CustodyPage />, store);

    await waitFor(() => {
      expect(container).toMatchSnapshot();
    });
  });

  it('opens connect custody without any custody selected', async () => {
    const { getByTestId } = renderWithProvider(<CustodyPage />, store);

    await waitFor(() => {
      expect(getByTestId('custody-connect-button')).toBeDefined();
    });
  });

  it('calls getCustodianJwtList on custody select when connect btn is click', async () => {
    const { getByTestId } = renderWithProvider(<CustodyPage />, store);

    const custodyBtn = getByTestId('custody-connect-button');
    await waitFor(() => {
      fireEvent.click(custodyBtn);
    });

    await waitFor(() => {
      expect(mockedGetCustodianJWTList).toHaveBeenCalled();
    });
  });

  it('clicks connect button and shows the jwt form', async () => {
    const { getByTestId } = renderWithProvider(<CustodyPage />, store);
    const custodyBtn = getByTestId('custody-connect-button');

    await waitFor(() => {
      fireEvent.click(custodyBtn);
    });

    await waitFor(() => {
      expect(screen.getByTestId('jwt-form-connect-button')).toBeInTheDocument();
    });
  });
});
