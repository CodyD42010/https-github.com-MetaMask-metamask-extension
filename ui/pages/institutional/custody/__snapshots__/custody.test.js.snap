// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustodyPage renders CustodyPage 1`] = `
<div>
  <div
    class="box box--flex-direction-row"
  >
    
    
    <div
      class="box box--sm:padding-7 box--md:padding-2 box--display-flex box--flex-direction-column"
    >
      <button
        aria-label="Back"
        class="box mm-button-icon mm-button-icon--size-sm box--display-flex box--flex-direction-row box--justify-content-center box--align-items-center box--color-icon-default box--background-color-transparent box--rounded-lg"
      >
        <span
          class="box mm-icon mm-icon--size-sm box--display-inline-block box--flex-direction-row box--color-inherit"
          style="mask-image: url('./images/icons/arrow-left.svg');"
        />
      </button>
      <h4
        class="box mm-text mm-text--body-lg-medium box--margin-top-4 box--margin-bottom-4 box--flex-direction-row box--color-text-default"
      >
        Custodial Accounts
      </h4>
      <h6
        class="box mm-text mm-text--body-md box--margin-top-2 box--margin-bottom-5 box--flex-direction-row box--color-text-default"
      >
        Please choose the custodian you want to connect in order to add or refresh a token.
      </h6>
      <div
        class="box box--flex-direction-row"
      >
        <ul
          width="full"
        >
          <div
            class="box box--margin-bottom-4 box--padding-3 box--sm:padding-4 box--display-flex box--flex-direction-row box--justify-content-space-between box--align-items-center box--rounded-sm box--border-color-border-default box--border-style-solid box--border-width-1"
          >
            <div
              class="box box--display-flex box--flex-direction-row box--align-items-center"
            >
              <img
                alt="Saturn Custody"
                height="32"
                src="https://saturn-custody-ui.dev.metamask-institutional.io/saturn.svg"
                width="32"
              />
              <p
                class="box mm-text mm-text--body-md box--margin-left-2 box--flex-direction-row box--color-text-default"
              >
                Saturn Custody
              </p>
            </div>
            <button
              class="box mm-text mm-button-base mm-button-base--size-sm mm-button-primary mm-text--body-md box--padding-right-4 box--padding-left-4 box--display-inline-flex box--flex-direction-row box--justify-content-center box--align-items-center box--color-primary-inverse box--background-color-primary-default box--rounded-pill"
              data-testid="custody-connect-button"
            >
              Select
            </button>
          </div>
        </ul>
      </div>
    </div>
    
  </div>
</div>
`;

exports[`CustodyPage renders CustodyPage 2`] = `
<div>
  <div
    class="box box--flex-direction-row"
  >
    
    
    <div
      class="box box--sm:padding-7 box--md:padding-2 box--display-flex box--flex-direction-column"
    >
      <button
        aria-label="Back"
        class="box mm-button-icon mm-button-icon--size-sm box--display-flex box--flex-direction-row box--justify-content-center box--align-items-center box--color-icon-alternative box--background-color-transparent box--rounded-lg"
      >
        <span
          class="box mm-icon mm-icon--size-sm box--display-inline-block box--flex-direction-row box--color-inherit"
          style="mask-image: url('./images/icons/arrow-left.svg');"
        />
      </button>
      <h4
        class="box mm-text mm-text--body-md box--flex-direction-row box--color-text-default"
      >
        <div
          class="box box--display-flex box--flex-direction-row box--align-items-center"
        >
          <p
            class="box mm-text mm-text--body-md box--margin-left-2 box--flex-direction-row box--color-text-default"
          />
        </div>
      </h4>
      <p
        class="box mm-text mm-text--body-md box--margin-top-4 box--margin-bottom-4 box--flex-direction-row box--color-text-default"
      >
        Enter your  token or add a new token
      </p>
    </div>
    <div
      class="box box--padding-top-7 box--padding-bottom-7 box--flex-direction-row"
    >
      <div
        class="box jwt-url-form box--margin-bottom-8 box--display-flex box--flex-direction-column box--align-items-flex-start"
      >
        <div
          class="box jwt-url-form__jwt-container box--margin-top-4 box--margin-bottom-6 box--display-flex box--flex-direction-column box--align-items-center box--width-full"
        >
          <div
            class="box box--flex-direction-row"
          >
            <p
              class="box mm-text jwt-url-form__instruction mm-text--body-md box--display-block box--flex-direction-row box--color-text-default"
            >
              Paste or drop your token here:
            </p>
            <textarea
              class="jwt-url-form__input-jwt"
              data-testid="jwt-input"
              id="jwt-box"
            >
              token
            </textarea>
          </div>
        </div>
        <div
          class="box box--flex-direction-row box--width-full"
        >
          <p
            class="box mm-text jwt-url-form__instruction mm-text--body-md box--display-block box--flex-direction-row box--color-text-default"
          >
             API URL
          </p>
          <div
            class="box box--flex-direction-row"
          >
            <input
              class="jwt-url-form__input"
              data-testid="jwt-api-url-input"
              id="api-url-box"
              value="url"
            />
          </div>
        </div>
      </div>
      <div
        class="box box--padding-4 box--display-flex box--flex-direction-row box--justify-content-center"
      >
        <button
          class="box mm-text mm-button-base mm-button-base--size-md mm-button-primary mm-text--body-md box--margin-right-4 box--padding-right-4 box--padding-left-4 box--display-inline-flex box--flex-direction-row box--justify-content-center box--align-items-center box--color-primary-inverse box--background-color-primary-default box--rounded-pill"
          type="secondary"
        >
          Cancel
        </button>
        <button
          class="box mm-text mm-button-base mm-button-base--size-md mm-button-primary mm-text--body-md box--padding-right-4 box--padding-left-4 box--display-inline-flex box--flex-direction-row box--justify-content-center box--align-items-center box--color-primary-inverse box--background-color-primary-default box--rounded-pill"
          data-testid="jwt-form-connect-button"
        >
          Connect
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`CustodyPage renders CustodyPage 3`] = `<div />`;
